torchbraid/odenet_apps.py

```diff
@@ -230,20 +193,22 @@ def getPrimalWithGrad(self,tstart,tstop,level):
     so it can be stored internally instead of
     being recomputed.
     """
-    
-    layer = self.getLayer(tstart,tstop,level)
 
     b_x = self.getUVector(0,tstart)
-    t_x = b_x.tensor()
 
-    self.setLayerWeights(tstart,tstop,level,b_x.weightTensors())
+    ts_index = self.getGlobalTimeIndex(tstart)-self.start_layer
 
+    assert(ts_index<len(self.layer_models))
```

Comment:
Might be good to have an assert(ts_index >= 0) here too, just to be sure, and catch potential future bugs...